error installing cryptography module

<!-- ⚠️ If you do not respect this template, your issue will be closed -->
<!-- ⚠️ Make sure to browse the opened and closed issues -->

### Informations

- **Qiskit version 0.10.1**
- **Python version 3.5.2**
- **Operating system Ubuntu 16.04 LTS**

### What is the current behavior?
`pip install qiskit-ibmq-provider` fails to install with a compiler error during building `cryptography` module

```
  Failed building wheel for cryptography
  Running setup.py clean for cryptography
Failed to build cryptography
Installing collected packages: ntlm-auth, asn1crypto, six, pycparser, cffi, cryptography, urllib3, idna, chardet, certifi, requests, requests-ntlm, ply, marshmallow, marshmallow-polyfield, mpmath, sympy, psutil, numpy, scipy, decorator, networkx, pylatexenc, jsonschema, pillow, qiskit-terra, websockets, qiskit-ibmq-provider
  Running setup.py install for cryptography ... error
    Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-dy7d10o4/cryptography/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-9vewcth6-record/install-record.txt --single-version-externally-managed --compile --user --prefix=:
    /usr/lib/python3.5/distutils/dist.py:261: UserWarning: Unknown distribution option: 'python_requires'
      warnings.warn(msg)
    running install
    running build
    running build_py

...

    reading manifest file 'src/cryptography.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    no previously-included directories found matching 'docs/_build'
    warning: no previously-included files matching '*' found under directory 'vectors'
    writing manifest file 'src/cryptography.egg-info/SOURCES.txt'
    running build_ext
    generating cffi module 'build/temp.linux-x86_64-3.5/_padding.c'
    creating build/temp.linux-x86_64-3.5
    generating cffi module 'build/temp.linux-x86_64-3.5/_constant_time.c'
    generating cffi module 'build/temp.linux-x86_64-3.5/_openssl.c'
    building '_openssl' extension
    creating build/temp.linux-x86_64-3.5/build
    creating build/temp.linux-x86_64-3.5/build/temp.linux-x86_64-3.5
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/python3.5m -c build/temp.linux-x86_64-3.5/_openssl.c -o build/temp.linux-x86_64-3.5/build/temp.linux-x86_64-3.5/_openssl.o -Wconversion -Wno-error=sign-conversion
    build/temp.linux-x86_64-3.5/_openssl.c:498:30: fatal error: openssl/opensslv.h: No such file or directory
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    
    ----------------------------------------
Command "/usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-dy7d10o4/cryptography/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-9vewcth6-record/install-record.txt --single-version-externally-managed --compile --user --prefix=" failed with error code 1 in /tmp/pip-build-dy7d10o4/cryptography/
```

### Steps to reproduce the problem

`pip install qiskit-ibmq-provider` 


### What is the expected behavior?

correct installation of the library

### Suggested solutions

pre-req installation of `libssl-dev`
`sudo apt-get install libssl-dev`

Maybe this should be added to the documentation?

--- a/docs/faq.rst
+++ b/docs/faq.rst
@@ -53,3 +53,30 @@ environment. Choose a virtual environment in the Anaconda-Navigator from the
 see all of the virtual environments within Anaconda, and you can
 select the environment where you have Qiskit installed to launch Jupyter
 Notebook.
+
+|
+
+**Q: Why am I getting a compilation error while installing ``qiskit``?**
+
+**A:** Qiskit depends on a number of other open source Python packages, which
+are automatically installed when doing ``pip install qiskit``. Depending on
+your system's platform and Python version, is it possible that a particular
+package does not provide a pre-built binary wheel for your system, and in those
+cases ``pip`` will attempt to compile the package from source, which in turn
+might require some extra dependencies that need to be installed manually.
+
+If the output of ``pip install qiskit`` contains similar lines to:
+
+.. code:: sh
+
+  Failed building wheel for SOME_PACKAGE
+  ...
+  build/temp.linux-x86_64-3.5/_openssl.c:498:30: fatal error
+  compilation terminated.
+  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
+
+please check the documentation of the package that failed to install (in the
+example code, ``SOME_PACKAGE``) for information on how to install the libraries
+needed for compiling from source. For example:
+
+* ``cryptography``: https://cryptography.io/en/latest/installation/#building-cryptography-on-linux

